pkg/cmd/pulumi/install.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/logging"
+
+	"github.com/spf13/cobra"
+
+	"github.com/pulumi/pulumi/pkg/v3/backend/display"
+	"github.com/pulumi/pulumi/pkg/v3/engine"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+)
+
+func newInstallCmd() *cobra.Command {
+	var reinstall bool
+
+	cmd := &cobra.Command{
+		Use:   "install",
+		Args:  cmdutil.NoArgs,
+		Short: "Install packages and plugins for the current program",
+		Long: "Install packages and plugins for the current program.\n" +
+			"\n" +
+			"This command is used to manually install packages and plugins required by your program.",
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			ctx := commandContext()
+			displayOpts := display.Options{
+				Color: cmdutil.GetGlobalColorization(),
+			}
+
+			// Load the project
+			proj, root, err := readProject()
+			if err != nil {
+				return err
+			}
+
+			span := opentracing.SpanFromContext(ctx)
+			projinfo := &engine.Projinfo{Proj: proj, Root: root}
+			pwd, main, pctx, err := engine.ProjectInfoContext(
+				projinfo,
+				nil,
+				cmdutil.Diag(),
+				cmdutil.Diag(),
+				false,
+				span,
+				nil,
+			)
+			if err != nil {
+				return err
+			}
+
+			defer pctx.Close()
+
+			// First make sure the language plugin is present.  We need this to load the required resource plugins.
+			// TODO: we need to think about how best to version this.  For now, it always picks the latest.
+			runtime := proj.Runtime
+			lang, err := pctx.Host.LanguageRuntime(pctx.Root, pctx.Pwd, runtime.Name(), runtime.Options())
+			if err != nil {
+				return fmt.Errorf("failed to load language plugin %s: %w", runtime.Name(), err)
+			}
+
+			if err = lang.InstallDependencies(root); err != nil {
+				return fmt.Errorf("installing dependencies failed; rerun manually to try again, "+
```

Comment:
As someone with not a lot of experience with pulumi "rerun manually" is a bit confusing.  I would have thought `pulumi install` is already running the install manually.  Can we give some better instructions how to debug this/how to install dependencies manually here?